martor/widgets.py

```diff
@@ -32,12 +32,16 @@ def render(self, name, value, attrs=None, renderer=None, **kwargs):
         # Make the settings the default attributes to pass
         attributes_to_pass = {
             "data-enable-configs": MARTOR_ENABLE_CONFIGS,
-            "data-upload-url": MARTOR_UPLOAD_URL,
-            "data-markdownfy-url": MARTOR_MARKDOWNIFY_URL,
-            "data-search-users-url": MARTOR_SEARCH_USERS_URL,
-            "data-base-emoji-url": MARTOR_MARKDOWN_BASE_EMOJI_URL,
+            "data-markdownfy-url": reverse("martor_markdownfy"),
         }
 
+        if MARTOR_UPLOAD_URL:
+            attributes_to_pass["data-upload-url"] = reverse("imgur_uploader")
+        if MARTOR_SEARCH_USERS_URL:
+            attributes_to_pass["data-search-users-url"] = reverse("search_user_json")
+        if MARTOR_SEARCH_USERS_URL:
+            attributes_to_pass["data-base-emoji-url"] = MARTOR_MARKDOWN_BASE_EMOJI_URL
```

Comment:
This was the main problem here: some endpoints can be configured in `settings.py` but will only be respected by the front. The backend doesn't use the values in `settings.py` to create paths so there is an inconsistency between JS and Django.

This also allows to include martor URLs under a sub path and they will correctly be resolved.